### ISSUE TITLE
No error on `super.` for inferred class properties in JavaScript

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#55884]
In https://github.com/microsoft/TypeScript/pull/54056 we added checks for class fields; however, we don't issue the same error in JavaScript for properties inferred from assignments.

```js
// @ts-check

class YaddaBase {
    constructor() {
        this.roots = "hi";
    }
}

class DerivedYadda extends YaddaBase {
    get rootTests() {
        return super.roots; // no error?
    }
}

console.log(new DerivedYadda().rootTests); // prints undefined!
```

_Originally posted by @DanielRosenwasser in https://github.com/microsoft/TypeScript/issues/55883#issuecomment-1736293678_
